### DISCUSSION TITLE
Ordered lists with letters???

### DISCUSSION THREAD

**Discussion by technovangelist** [markdoc/markdoc#240] [Q&A] [answered]
How do I do an ordered list but with letters instead of numbers? Normally I could do this: 
`<ol type="a">
  <li>Coffee</li>
  <li>Tea</li>
  <li>Milk</li>
</ol>`

but not here.

**Comment by segphault** [comment, marked as answer]
Markdown doesn't natively support using letters for ordered lists. Typically, in HTML you would set the [`type`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol#attr-type) attribute on the `ol` tag in order to get it to use lower case letters (`<ol type="a">...</ol>`).

To achieve something comparable in Markdoc, what you can do is define a custom `list` tag that accepts a `type` attribute and applies it to an enclosed ordered list:

```javascript
const output = Markdoc.transform(doc, {
  tags: {
    list: {
      attributes: {
        type: { type: 'string' },
      },
      transform(node, config) {
        const children = node.transformChildren(config);
        if (children.length && children[0].name === 'ol')
          children[0].attributes.type = node.attributes.type;
        return children;
      },
    },
  },
});
```

And then you would use it like this:

```
{% list type="a" %}
1. foo
2. bar
3. baz
{% /list %}
